modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestHandler.java

```diff
@@ -64,7 +64,13 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
                         Unpooled.copiedBuffer(request.content()),
                         request.headers(),
                         request.trailingHeaders());
-        final Netty4HttpRequest httpRequest = new Netty4HttpRequest(serverTransport.xContentRegistry, copy, ctx.channel());
+        final Netty4HttpRequest httpRequest;
+        try {
+            httpRequest = new Netty4HttpRequest(serverTransport.xContentRegistry, copy, ctx.channel());
+        } catch (Exception ex) {
+            request.release();
```

Comment:
I don't think this is right, I think this should only be released here if we added a retain which only happens if `msg` is an instance of `HttpPipelinedRequest`.